src/core/geometry/qgsgeometry.h

```diff
@@ -1552,23 +1552,135 @@ class CORE_EXPORT QgsGeometry
     % End
 #endif
 
+#ifndef SIP_RUN
+
     /**
-     * Returns contents of the geometry as a multi point
-     * if wkbType is WKBMultiPoint, otherwise an empty list
+     * Returns the contents of the geometry as a multi-point.
+     *
+     * Any z or m values present in the geometry will be discarded.
+     *
+     * \warning If the geometry is not a multi-point type, an empty list will be returned.
      */
     QgsMultiPointXY asMultiPoint() const;
+#else
+
+    /**
+     * Returns the contents of the geometry as a multi-point.
+     *
+     * Any z or m values present in the geometry will be discarded.
+     *
+     * This method works only with multi-point geometry types. If the geometry
+     * is not a multi-point type, a TypeError will be raised. If the geometry is null, a ValueError
+     * will be raised.
+     */
+    SIP_PYOBJECT asMultiPoint() const SIP_TYPEHINT( QgsMultiPointXY );
+    % MethodCode
+    const QgsWkbTypes::Type type = sipCpp->wkbType();
+    if ( sipCpp->isNull() )
+    {
+      PyErr_SetString( PyExc_ValueError, QStringLiteral( "Null geometry cannot be converted to a multipoint." ).toUtf8().constData() );
```

Comment:
Should this be translated?

```
try:
except ValueError as e:
    QMessageBox.warning( '[Translated generic error message goes here]: "{}"'.format( e.msg ) )
```

Not completely sure here. But since I have seen even devs work with translated linux terminals, I guess this should be translated too.

Reply:
Could you try getting a UnicodeEncodeError with your localised python and see if that's localised? We should just follow python's standard approach I think.

Reply:
Ok, seems cpython does not have localised exceptions -- they are all English. I think we should follow suit.